src/services/expenses.js

```diff
@@ -0,0 +1,76 @@
+const { v4: uuidv4 } = require('uuid');
+const usersService = require('../services/user');
+
+let Expenses = [];
+
+const getAll = (userId, categories, from, to) => {
+  let filteredExpenses = [...Expenses];
+
+  if (userId) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.userId === +userId,
+    );
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses.filter(
+      (expense) => expense.category === categories,
+    );
+  }
+
+  if (from && to) {
+    const dateFrom = new Date(from);
+    const dateTo = new Date(to);
+
+    filteredExpenses = filteredExpenses.filter(
+      (expense) =>
+        dateFrom <= new Date(expense.spentAt) &&
+        new Date(expense.spentAt) <= dateTo,
+    );
+  }
```

Comment:
you use filter at least three times here. would be good to combine it to iterate only once